soc/nordic/ironside/include/nrf_ironside/bootmode.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_NORDIC_IRONSIDE_INCLUDE_NRF_IRONSIDE_BOOTMODE_H_
+#define ZEPHYR_SOC_NORDIC_IRONSIDE_INCLUDE_NRF_IRONSIDE_BOOTMODE_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stddef.h>
+
+/**
+ * @name Boot mode service error codes.
+ * @{
+ */
+
+/** Invalid/unsupported boot mode transition. */
+#define IRONSIDE_BOOTMODE_ERROR_UNSUPPORTED_MODE (1)
+/** Failed to reboot into the boot mode due to other activity preventing a reset. */
+#define IRONSIDE_BOOTMODE_ERROR_BUSY (2)
+/** The boot message is too large to fit in the buffer. */
+#define IRONSIDE_BOOTMODE_ERROR_MESSAGE_TOO_LARGE (10)
+
+/**
+ * @}
+ */
+
+/* IronSide call identifiers with implicit versions. */
+#define IRONSIDE_CALL_ID_BOOTMODE_SERVICE_V1 5
```

Comment:
If there are more services with different IDs there could be good to prepare global header file and enum with IDs and refer to them